python/lsst/pipe/base/pipelineIR.py

```diff
@@ -40,6 +48,37 @@ class KeepInstrument:
     pass
 
 
+def standardize_uri_in_pipeline(path: str, directory: Optional[ButlerURI]) -> ButlerURI:
+    """Standardize a URI inside a pipeline description by expanding environment
+    variables and interpreting (or rejecting) relative paths.
+
+    Parameters
+    ----------
+    path : `str`
+        Path to standardize.
+    directory : `ButlerURI` or `None`
+        Directory URI that relative paths should be considered relative to.
+
+    Returns
+    -------
+    uri : `ButlerURI`
+        Standardized URI.
+
+    Raises
+    ------
+    RuntimeError
+        Raised if ``path`` is relative but ``directory`` is `None`.
+    """
+    # ButlerURI will expand variables automatically.
+    uri = ButlerURI(path, forceAbsolute=False)
+    if not uri.isabs():
+        if directory is not None:
+            uri = directory.join(uri)
```

Comment:
Are there any checks in ResourcePath to ensure URI types are not mixed? like if path gets interpreted as filelike, but directory is WebDav? I have to imagine there is, but wanted to highlight to check.